src/metadata.cpp

```diff
@@ -796,10 +796,12 @@ Metadata::SchemaSnapshot Metadata::schema_snapshot(int protocol_version, const V
 }
 
 void Metadata::update_keyspaces(int protocol_version, const VersionNumber& cassandra_version, ResultResponse* result) {
+  // update_keyspaces may race with update_partitions.
+  // Obtain lock via mutex_ to protect schema_snapshot_version_.
+  ScopedMutex l(&mutex_);
```

Comment:
Please undo the changes. `Metadata::update_keyspaces` looks good (as is).
(I think `update_partitions` should be fixed instead. See other comment..)